src/main/scala/mesosphere/marathon/api/v2/MesosResource.scala

```diff
@@ -0,0 +1,24 @@
+package mesosphere.marathon.api.v2
+
+import javax.ws.rs.core.{ MediaType, Response }
+import javax.ws.rs.{ GET, Path, Produces }
+
+import com.google.inject.Inject
+import mesosphere.marathon.MarathonConf
+import mesosphere.util.state.MesosMasterUtil
+
+@Path("v2/mesos")
+class MesosResource @Inject() (
+    mesosMasterUtil: MesosMasterUtil,
+    conf: MarathonConf) {
+  @GET
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  def index(): Response = {
```

Comment:
This is mostly already included in the `InfoResource`.

Just add the `"masterUrl"` field as `"mesos_master_url"`.